controllers/datafoundation.go

```diff
@@ -128,7 +128,7 @@ func dfReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *
 	// Set up the datafoundation sub-reconciler
 	r := dataFoundationReconciler{}
 	r.initReconciler(offeringReconciler, offering)
-	r.spec = offeringSpec.(dataFoundationSpec)
+	r.spec = *offeringSpec.(*dataFoundationSpec)
```

Comment:
```suggestion
	r.spec, ok = *offeringSpec.(*dataFoundationSpec)
	if !ok {
	    return ctrl.Result{}, fmt.Errorf("Error while typecasting into dataFoundationSpec)
	 }	
```
I think we should catch the error and return otherwise the code will panic. Please feel free to update the error message to something more meaningful. Same comment for `dfcReconcile`

Reply:
@nb-ohad suggested that if there is an error typecasting we should panic

Reply:
And what was the reason for that?
If we panic then the pod will go into CLBO, if we don't then we will have a proper error mentioning there is something wrong with the `spec.config` section.

Reply:
Yes, I agree having a error message instead of code panic and pod running to CLBO will be better

Reply:
@bindrad 
if there is a panic here it means that the developer made the mistake of using one type in one place of the code while using another type here. This is a coding mistake, not a runtime error. If we could catch that in compile or lint I would prefer that, but as we cannot, the next best thing is a panic as it is the most obvious way to get the developer's attention, and will be consistent on every invocation!

Reply:
updated

Reply:
Okay, got it. Thanks!